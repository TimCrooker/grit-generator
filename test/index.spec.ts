import { Projen } from "projenerator"
import path from 'path'

test('Generator cascade', async () => {
	// run the top level generator as not a mock and direct output to local fixtures
	const genGen = new Projen({
		generator: path.join(__dirname, "../"),
		mock: true,
	})
	await genGen.run()
	expect(await genGen.getOutputFiles()).toMatchSnapshot()
})
