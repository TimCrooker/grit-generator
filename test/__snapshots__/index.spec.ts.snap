// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generator variations Should render with plugins 1`] = `
Array [
  ".gitignore",
  "LICENSE",
  "README.md",
  "babel.config.js",
  "generator.ts",
  "jest.config.js",
  "package.json",
  "plugins/plugin1/foo.txt",
  "plugins/plugin1/pluginFile.ts",
  "plugins/plugin2/bar.json",
  "plugins/plugin2/pluginFile.ts",
  "template/LICENSE",
  "template/README.md",
  "template/_package.json",
  "template/gitignore",
  "test/test.spec.ts",
  "tsconfig.json",
]
`;

exports[`Generator variations Should render without plugins 1`] = `
Array [
  ".gitignore",
  "LICENSE",
  "README.md",
  "babel.config.js",
  "generator.ts",
  "jest.config.js",
  "package.json",
  "plugins/plugin1/foo.txt",
  "plugins/plugin1/pluginFile.ts",
  "plugins/plugin2/bar.json",
  "plugins/plugin2/pluginFile.ts",
  "template/LICENSE",
  "template/README.md",
  "template/_package.json",
  "template/gitignore",
  "test/test.spec.ts",
  "tsconfig.json",
]
`;
